traversal/scanner/GraphIterator.java

```diff
@@ -400,10 +397,9 @@ private void clearScopes() {
         }
 
         private Forwardable<Vertex<?, ?>, Order.Asc> createIteratorFromEdges() {
-            List<Forwardable<Vertex<?, ?>, Order.Asc>> iterators = new ArrayList<>();
-            procedureVertex.ins().forEach(edge -> iterators.add(branch(vertexTraversers.get(edge.from()).vertex(), edge)));
-            if (iterators.size() == 1) return iterators.get(0);
-            else return intersect(iterate(iterators), ASC);
+            Map<ProcedureEdge<?,?>, Forwardable<Vertex<?, ?>, Order.Asc>> iterators = new HashMap<>();
+            procedureVertex.ins().forEach(edge -> iterators.put(edge, branch(vertexTraversers.get(edge.from()).vertex(), edge)));
+            return procedureVertex.intersect(iterators, params, ASC);
```

Comment:
If the vertex is a Value procedure vertex:
          1. Find the expression input edges, get the input edges' values, compute the output value
          2. apply all remaining edges as predicates using isClosures instead of branch()

          else:
          1. open all the iterators with Branch
          2. intersect them all
          
          this should delete `ValueVertexSpec` and also remove the need for special intersection operations

Reply:
Weren't we doing away with isClosure?